test/heapwatch/metrics_viz.py

```diff
@@ -93,17 +93,28 @@ def parse_metrics(fin: Iterable[str], nick: str, metrics_names: set=None) -> Tup
     except:
         print(f'An exception occurred in parse_metrics: {sys.exc_info()}')
         pass
+    if diff and metrics_names and len(metrics_names) == 2 and len(out) == 2:
+        m = list(out.keys())
+        name = f'{m[0]}_-_{m[1]}'
+        new_out = {name: out[m[0]] - out[m[1]]}
+        new_types = {name: TYPE_GAUGE}
+        out = new_out
+        types = new_types
+
     return out, types
 
 
 def main():
     os.environ['TZ'] = 'UTC'
     time.tzset()
+    default_output_file = 'metrics_viz.png'
 
     ap = argparse.ArgumentParser()
     ap.add_argument('metrics_names', nargs='+', default=None, help='metric name(s) to track')
     ap.add_argument('-d', '--dir', type=str, default=None, help='dir path to find /*.metrics in')
     ap.add_argument('-l', '--list-nodes', default=False, action='store_true', help='list available node names with metrics')
+    ap.add_argument('-s', '--save', action='store_true', default=None, help=f'save plot to \'{default_output_file}\' file instead of showing it')
+    ap.add_argument('--diff', action='store_true', default=None, help='diff two metrics instead of plotting their values')
```

Comment:
Note: Diffing two arbitrary metrics will not work well, they would have to be in the same units. Perhaps we should mention this as a concern.

Reply:
added